### PR TITLE
HTTP: Refactor Call* helpers to CallFromRequest

### PR DESCRIPTION
**Pull request by JensSkipr**

### CODE REVIEW

File: http/http.go [hunk 1]
```diff
@@ -34,71 +69,172 @@ import (
...
-			return nil, errors.NewGenericError(500, errorDomain, errorSubDomain, ErrorMarshalRequestBodyFailed, nil)
+	if req.Body != nil {
+		switch reqType {
+		case RequestTypeJSON:
+			if _, ok := req.Headers["Content-Type"]; !ok {
+				req.Headers["Content-Type"] = "application/json"
+			}
+			bodyBytes, err = json.Marshal(req.Body)
+			if err != nil {
+				defaultLog.WithField("error", err).Error("Failed to marshal request body to JSON")
```
Comment (florianjeanmart) [inline http/http.go:98]: Could you add the req.Body as field of the log ?

File: http/http.go [hunk 2]
```diff
@@ -34,71 +69,172 @@ import (
...
+				return nil, errors.NewGenericError(500, errorDomain, errorSubDomain, ErrorUnsupportedFieldType, meta)
+			}
+
+		case RequestTypeXML:
+			if _, ok := req.Headers["Content-Type"]; !ok {
+				req.Headers["Content-Type"] = "application/xml"
+			}
+			bodyBytes, err = xml.Marshal(req.Body)
+			if err != nil {
+				defaultLog.WithField("error", err).Error("Failed to marshal request body to XML")
```
Comment (florianjeanmart) [inline http/http.go:123]: Could you add the req.Body as field of the log ?

File: http/http.go [hunk 3]
```diff
@@ -13,12 +15,45 @@ import (
...
-// Call marshals the body to JSON and sends a HTTP request. Response is parsed as JSON.
+type RequestType string
+
+const (
+	RequestTypeJSON RequestType = "JSON"
+	RequestTypeRaw  RequestType = "RAW"
+	RequestTypeXML  RequestType = "XML"
+)
+
+type Request struct {
```
Comment (florianjeanmart) [inline http/http.go:26]: Could you add a comment to specify the optional / required fields ?

**Review by florianjeanmart — APPROVED** [review]
LGFM. Just some comments about documentaton and logging

File: http/http.go (hunk 1 above)
Comment (JensSkipr) [reply http/http.go:98]: The body is already included in the log, see below code. It's one of the nice tricks you can do with Logrus.
```go
	defaultLog := log.WithFields(log.Fields{
		"method":  req.Method,
		"url":     req.RootURL + req.Path,
		"body":    req.Body,
		"query":   req.Query,
		"headers": req.Headers,
	})
```

File: http/http.go (hunk 2 above)
Comment (JensSkipr) [reply http/http.go:123]: See https://github.com/skiprco/go-utils/pull/52#discussion_r603100345

File: http/http.go (hunk 1 above)
Comment (florianjeanmart) [reply http/http.go:111]: Ok, nice trick !

File: http/http.go (hunk 3 above)
Comment (JensSkipr) [reply http/http.go:26]: Fixed by 69b44db

**Pull request merged**